experimental/dds/tree2/src/core/tree/detachedFieldIndex.ts

```diff
@@ -141,29 +157,105 @@ export class DetachedFieldIndex {
 	}
 
 	public encode(): string {
-		const data: [Major, Minor, ForestRootId][] = [];
-		forEachInNestedMap(this.detachedNodeToField, (root, key1, key2) => {
-			data.push([key1, key2, root]);
-		});
-		return JSON.stringify({
-			data,
-			id: this.rootIdAllocator.getNextId(),
-		});
+		return this.codec.encode({
+			data: this.detachedNodeToField,
+			maxId: this.rootIdAllocator.getNextId(),
+		}) as string;
 	}
 
 	/**
 	 * Loads the tree index from the given string, this overrides any existing data.
 	 */
 	public loadData(data: string): void {
-		const detachedFieldIndex: { data: readonly [Major, Minor, ForestRootId][]; id: number } =
-			JSON.parse(data);
-		const map = new Map();
-		for (const [major, minor, root] of detachedFieldIndex.data) {
-			setInNestedMap(map, major, minor, root);
-		}
-		this.detachedNodeToField = map;
+		const detachedFieldIndex: {
+			data: NestedMap<Major, Minor, ForestRootId>;
+			maxId: number;
+		} = this.codec.decode(data);
+
 		this.rootIdAllocator = idAllocatorFromMaxId(
-			detachedFieldIndex.id,
+			detachedFieldIndex.maxId,
 		) as IdAllocator<ForestRootId>;
+		this.detachedNodeToField = detachedFieldIndex.data;
 	}
 }
+
+// Define the Major and Minor types:
+const MajorSchema = Type.Union([Type.String(), Type.Number(), Type.Undefined()]);
```

Comment:
Could you organize the format types and codec into their own file? Structure should match e.g. how modular change works:

`detachedFieldIndexFormat` for typebox schema declarations / encoded format types
`detachedFieldIndexTypes` for in-memory types (this is probably just the type `{
			data: NestedMap<Major, Minor, ForestRootId>;
			maxId: ForestRootId;
		}`, which you should name e.g. `DetachedFieldSummaryData`)
`detachedFieldIndexCodec` for the codec implementation

Reply:
Organized format types into different files